android/app/src/main/java/org/fossasia/openevent/viewmodels/SearchActivityViewModel.java

```diff
@@ -0,0 +1,111 @@
+        realm = Realm.getDefaultInstance();
+    }
+
+    public LiveData<List<Object>> getSearchResults() {
+        objectList.clear();
+        addResultFromSessions(searchText);
+        addResultsFromLocations(searchText);
+        addResultsFromSpeakers(searchText);
+        addResultsFromTracks(searchText);
+
+        searchResults.setValue(objectList);
+        return searchResults;
+    }
+
+    public String getSearchText() {
+        return searchText;
+    }
+
+    public void setSearchText(String searchText) {
+        String wildcardQuery = String.format("*%s*", searchText.toLowerCase(Locale.getDefault()));
+        this.searchText = wildcardQuery;
+    }
+
+    public void addResultsFromTracks(String queryString) {
+        RealmResults<Track> filteredTracks = realm.where(Track.class)
+                .like("name", queryString, Case.INSENSITIVE).findAllSortedAsync("name");
+
+        filteredTracks.addChangeListener(tracks -> {
+            if (tracks.size() > 0) {
+                objectList.add("Tracks");
+            }
+            objectList.addAll(tracks);
+            Timber.d("Filtering done total results %d", tracks.size());
+        });
+    }
+
+    public void addResultsFromSpeakers(String queryString) {
+        RealmResults<Speaker> filteredSpeakers = realm.where(Speaker.class)
+                .like("name", queryString, Case.INSENSITIVE).or()
+                .like("country", queryString, Case.INSENSITIVE).or()
+                .like("organisation", queryString, Case.INSENSITIVE).findAllSortedAsync("name");
+
+        filteredSpeakers.addChangeListener(speakers -> {
+            if (speakers.size() > 0) {
+                objectList.add("Speakers");
+            }
+            objectList.addAll(speakers);
+            Timber.d("Filtering done total results %d", speakers.size());
+        });
+    }
+
+    public void addResultsFromLocations(String queryString) {
+        RealmResults<Microlocation> filteredMicrolocations = realm.where(Microlocation.class)
+                .like("name", queryString, Case.INSENSITIVE).findAllSortedAsync("name");
+
+        filteredMicrolocations.addChangeListener(microlocations -> {
+            if (microlocations.size() > 0) {
+                objectList.add("Locations");
+            }
+            objectList.addAll(microlocations);
+            Timber.d("Filtering done total results %d", microlocations.size());
+        });
+    }
+
+    public void addResultFromSessions(String queryString) {
+        RealmResults<Session> filteredSessions = realm.where(Session.class)
+                .like("title", queryString, Case.INSENSITIVE).findAllSortedAsync("title");
+
+        filteredSessions.addChangeListener(sessions -> {
+            if (sessions.size() > 0) {
+                objectList.add("Sessions");
+            }
+            objectList.addAll(sessions);
+            Timber.d("Filtering done total results %d", sessions.size());
+        });
+    }
+
+    public void onDestroy(){
```

Comment:
This should be overriden method, `onCleared`

Reply:
@iamareebjamal Done the change!